src/wp-includes/class-wp-query.php

```diff
@@ -2020,8 +2020,8 @@ public function get_posts() {
 		}
 
 		if ( isset( $q['page'] ) ) {
-			$q['page'] = trim( $q['page'], '/' );
-			$q['page'] = absint( $q['page'] );
+			$q['page'] = is_scalar( $q['page'] ) ? trim( $q['page'], '/' ) : '';
+			$q['page'] = is_scalar( $q['page'] ) ? absint( $q['page'] ) : 0;
```

Comment:
It appears that the second `is_scalar()` check is redundant here, as `$q['page']` will always be a string after the previous line, so I think line 2024 can stay as is:
```suggestion
			$q['page'] = absint( $q['page'] );
```